node/src/main.rs

```diff
@@ -323,186 +256,58 @@ async fn main() {
-                )
-            })
-            .collect::<BTreeMap<String, ChainIdentifier>>();
-
-        // Note that both `eth_firehose_only_idents` and `ethereum_idents` contain Ethereum
-        // networks. If the same network is configured in both RPC and Firehose, the RPC ident takes
-        // precedence. This is necessary because Firehose endpoints currently have no `net_version`.
-        // See also: firehose-no-net-version.
-        let mut network_identifiers = eth_firehose_only_idents;
-        network_identifiers.extend(ethereum_idents);
-        network_identifiers.extend(arweave_idents);
-        network_identifiers.extend(near_idents);
-        network_identifiers.extend(cosmos_idents);
-        network_identifiers.extend(substream_idents);
-        network_identifiers.extend(starknet_idents);
-
-        let network_store = store_builder.network_store(network_identifiers);
-
-        let arweave_chains = networks_as_chains::<arweave::Chain>(
-            &env_vars,
-            &mut blockchain_map,
-            &logger,
-            &arweave_networks,
-            substreams_networks_by_kind.get(&BlockchainKind::Arweave),
-            network_store.as_ref(),
-            &logger_factory,
-            metrics_registry.clone(),
-        );
-
-        let eth_firehose_only_networks = if eth_firehose_only_networks.networks.len() == 0 {
-            None
-        } else {
-            Some(&eth_firehose_only_networks)
-        };
-
-        if !opt.disable_block_ingestor && eth_networks.networks.len() != 0 {
-            let eth_network_names = Vec::from_iter(eth_networks.networks.keys());
-            let fh_only = match eth_firehose_only_networks {
-                Some(firehose_only) => Some(Vec::from_iter(firehose_only.networks.keys())),
-                None => None,
-            };
-            network_store
-                .block_store()
-                .cleanup_ethereum_shallow_blocks(eth_network_names, fh_only)
-                .unwrap();
+        // see comment on cleanup_ethereum_shallow_blocks
+        if !opt.disable_block_ingestor {
+            match blockchain_map
+                .get_all_by_kind::<graph_chain_ethereum::Chain>(BlockchainKind::Ethereum)
+                .ok()
+                .map(|chains| {
+                    chains
+                        .iter()
+                        .flat_map(|c| {
+                            if !c.chain_client().is_firehose() {
+                                Some(c.name.to_string())
+                            } else {
+                                None
+                            }
+                        })
+                        .collect()
+                }) {
+                Some(eth_network_names) => {
+                    network_store
+                        .block_store()
+                        .cleanup_ethereum_shallow_blocks(eth_network_names)
+                        .unwrap();
+                }
+                None => todo!(),
```

Comment:
Will this ever be reached? Is unreachable!() more suitable here?

Reply:
yep good catch I've updated and added an explanation